compiler-rt/lib/mpk_untrusted/mpk_formatter.cpp

```diff
@@ -1,42 +1,116 @@
 #include "mpk_formatter.h"
+#include "alloc_site_handler.h"
+#include <bits/stdint-intn.h>
+#include <bits/stdint-uintn.h>
+#include <fstream>
 
 namespace __mpk_untrusted {
 
-// Flush Allocs is to be called on program exit to flush all faulting
-// allocations to disk/file.
-void flushAllocs() {
-  auto handler = AllocSiteHandler::init();
+#define ATTEMPTS 128
 
-  std::vector<llvm::json::Value> jAllocVec;
-  for (auto alloc : handler->faultingAllocs()) {
-    jAllocVec.push_back(toJSON(alloc));
+bool makeUniqueFile(std::string Model, std::ofstream &OS) {
+  srand(time(NULL));
+  // Loop for number of attempts just in case level of entropy is to low
+  for (uint8_t attempt = 0; attempt != ATTEMPTS; ++attempt) {
+    std::string unique = Model;
+    for (uint32_t i = 0, e = unique.length(); i != e; ++i) {
+      if (unique[i] == '%')
+        unique[i] = "0123456789abcdef"[rand() & 15];
+    }
+    struct stat info;
+    // If file does not already exist, create and return ofstream
+    if (stat(unique.c_str(), &info) == -1) {
```

Comment:
Its going to call stat anyway under the hood, but you could just use `std::experimental::filesystem::exists()`. This code isn't wrong, but its a bit strange to mix stat and fstream.

This might also compose better if file name generation was a separate function.

you could also just return an optional file stream instead of bool, which may be more ergonomic.

Reply:
filesystem is probably overkill, and might be hard to link, even though clang is c++14. so disregard that. stat +fstream is a bit unorthodox though

Reply:
Changes made for makeUniqueFile address these concerns as well.